File: Practica4/Figuras.py
import math
# Círculo
def area(figura, **kwargs):
    if figura == "círculo":
        radio = kwargs.get('radio', 0)
        return math.pi * radio**2
    
    #Rectángulo
    elif figura == "rectángulo":
        largo = kwargs.get('largo', 0)
        ancho = kwargs.get('ancho', 0)
        return largo * ancho
    
    #Triángulo Rectángulo
    elif figura == "triángulo rectángulo":
        base = kwargs.get('base', 0)
        altura = kwargs.get('altura', 0)
        return (base * altura) / 2
    
    #Trapecio
    elif figura == "trapecio":
        base_mayor = kwargs.get('base_mayor', 0)
        base_menor = kwargs.get('base_menor', 0)
        altura = kwargs.get('altura', 0)
        return ((base_mayor + base_menor) * altura) / 2
    
    #Pentágono
    elif figura == "pentágono":
        apotema = kwargs.get('apotema', 0)
        lado = kwargs.get('lado', 0)
        return (5 * lado * apotema) / 2
    
    else:
        return "Figura desconocida"

print("Área del círculo:", area("círculo", radio=5))
print("Área del rectángulo:", area("rectángulo", largo=4, ancho=6))
print("Área del triángulo rectángulo:", area("triángulo rectángulo", base=3, altura=4))
print("Área del trapecio:", area("trapecio", base_mayor=8, base_menor=5, altura=3))
print("Área del pentágono:", area("pentágono", apotema=4, lado=6))

File: Practica4/Figuras/src/Prac4/Figuras.java
package Prac4;

public class Figuras {

    // Sobrecarga para el área del círculo
    public static double calcularArea(double radio) {
        return Math.PI * radio * radio;
    }

    // Sobrecarga para el área del rectángulo
    public static double calcularArea(double base, double altura) {
        return base * altura;
    }

    // Sobrecarga para el área del triángulo
    public static double calcularArea(double base, double altura, String tipo) {
        if (tipo.equalsIgnoreCase("triangulo")) {
            return (base * altura) / 2;
        }
        return -1;
    }

    // Sobrecarga para el área del trapecio
    public static double calcularArea(double baseMayor, double baseMenor, double altura, String tipo) {
        if (tipo.equalsIgnoreCase("trapecio")) {
            return ((baseMayor + baseMenor) * altura) / 2;
        }
        return -1; 
    }

    // Sobrecarga para el área del pentágono
    public static double calcularArea(double lado, double apotema, int lados) {
        if (lados == 5) {
            return (5 * lado * apotema) / 2;
        }
        return -1; 
    }

    public static void main(String[] args) {
        System.out.println("Área del círculo: " + calcularArea(5));  // Círculo
        System.out.println("Área del rectángulo: " + calcularArea(4, 6));  // Rectángulo
        System.out.println("Área del triángulo: " + calcularArea(4, 6, "triangulo"));  // Triángulo
        System.out.println("Área del trapecio: " + calcularArea(8, 4, 5, "trapecio"));  // Trapecio
        System.out.println("Área del pentágono: " + calcularArea(6, 4, 5));  // Pentágono
    }
}
